src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -522,6 +522,41 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 			break;
 		}
 
+		case T_CreateStatsStmt:
+		{
+			CreateStatsStmt *createStatsStmt = (CreateStatsStmt *) parseTree;
+
+			/* because CREATE STATISTICS statements can only have one relation */
+			RangeVar *relation = linitial(createStatsStmt->relations);
+
+			char **relationName = &(relation->relname);
+			char **objectSchemaName = &(relation->schemaname);
+
+			SetSchemaNameIfNotExist(objectSchemaName, schemaName);
+			AppendShardIdToName(relationName, shardId);
+
+			Value *statNameValue = llast(createStatsStmt->defnames);
+			char *statName = strVal(statNameValue);
+			char **statNameVar = &(statName);
+			AppendShardIdToName(statNameVar, shardId);
+			statNameValue = makeString(pstrdup(statName));
+
+			bool containsSchemaName = list_length(createStatsStmt->defnames) > 1;
+
+			if (containsSchemaName)
+			{
+				createStatsStmt->defnames = list_make2(linitial(
+														   createStatsStmt->defnames),
+													   statNameValue);
+			}
+			else
+			{
+				createStatsStmt->defnames = list_make1(statNameValue);
```

Comment:
Can the code ever come here? If so why did we not need to add the schema here too?

Reply:
No, it seems the code will never reach here, added this part just to be defensive. Didn't add a schema name because we don't have one, in this case

Reply:
For this whole block and the above, I would again do the transformations between `NameList` -> `RangeVar` -> `append shard id name in range var` -> convert back to `NameList`.

It also solves the unreachable parts of the code. And shows more the intention of the code.